HumanSpaceProgram/Assets/HSP.CelestialBodies/GravityUtils.cs

```diff
@@ -1,34 +1,54 @@
-﻿using UnityEngine;
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
 
 namespace HSP.CelestialBodies
 {
     public static class GravityUtils
     {
-        /// <summary>
-        /// Gravitational constant G [N * kg^-2 * m^2].
-        /// </summary>
-        public const double G = 6.67430e-11;
+        public static IEnumerable<(CelestialBody, Vector3Dbl)> GetNBodyGravityAccelerations( Vector3Dbl absolutePosition )
+        {
+            // method to get the gravitational contribution of every CB individually, mostly for analysis and for auxiliary functionality.
+
+            (CelestialBody, Vector3Dbl)[] acc = new (CelestialBody, Vector3Dbl)[CelestialBodyManager.CelestialBodyCount];
+            int i = 0;
+            foreach( var body in CelestialBodyManager.CelestialBodies )
+            {
+                Vector3Dbl toBody = body.ReferenceFrameTransform.AbsolutePosition - absolutePosition;
+
+                double distanceSq = toBody.sqrMagnitude;
+                if( distanceSq == 0.0 )
```

Comment:
I'd compare distanceSq with double.Epsilon

Reply:
This is there pretty much solely to prevent bodies attracting themselves gravitationally, since the input collection includes *all* bodies for allocation reasons. It's clarified now.